presto-raptor/src/test/java/com/facebook/presto/raptor/storage/TestOrcFileRewriter.java

```diff
@@ -413,9 +430,258 @@ public void testUncompressedSize(boolean useOptimizedOrcWriter)
+        }
+
+        // Add a column
+        File newFile1 = new File(temporary, randomUUID().toString());
+        OrcFileInfo info = createFileRewriter(useOptimizedOrcWriter).rewrite(
+                getColumnTypes(ImmutableList.of(3L, 7L, 10L), ImmutableList.of(BIGINT, createVarcharType(20), DOUBLE)),
+                file,
+                newFile1,
+                new BitSet(5));
+        assertEquals(info.getRowCount(), 4);
+        assertEquals(readAllBytes(file.toPath()), readAllBytes(newFile1.toPath()));
+
+        // Drop a column
+        File newFile2 = new File(temporary, randomUUID().toString());
+        info = createFileRewriter(useOptimizedOrcWriter).rewrite(
+                getColumnTypes(ImmutableList.of(7L, 10L), ImmutableList.of(createVarcharType(20), DOUBLE)),
+                newFile1,
+                newFile2,
+                new BitSet(5));
+        assertEquals(info.getRowCount(), 4);
+        if (useOptimizedOrcWriter) {
+            // Optimized writer will keep the only column
+            OrcReader orcReader = new OrcReader(fileOrcDataSource(newFile2), ORC, new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE));
+            orcReader.getColumnNames().equals(ImmutableList.of("7"));
+        }
+        else {
+            // Apache writer does not drop columns
+            assertEquals(readAllBytes(newFile1.toPath()), readAllBytes(newFile2.toPath()));
+        }
+
+        // Add a column with the different ID with different type
+        File newFile3 = new File(temporary, randomUUID().toString());
+        info = createFileRewriter(useOptimizedOrcWriter).rewrite(
+                getColumnTypes(ImmutableList.of(7L, 10L, 13L), ImmutableList.of(createVarcharType(20), DOUBLE, createVarcharType(5))),
+                newFile2,
+                newFile3,
+                new BitSet(5));
+        assertEquals(info.getRowCount(), 4);
+        assertEquals(readAllBytes(newFile2.toPath()), readAllBytes(newFile3.toPath()));
+
+        // Get prepared for the final file; make sure it is accessible from storage manager
+        UUID uuid = randomUUID();
+        File newFile4 = getFileSystemPath(new File(dataDir, "data/storage"), uuid);
+        if (useOptimizedOrcWriter) {
+            // Apache ORC writer creates the file itself
+            newFile4.getParentFile().mkdirs();
+            newFile4.createNewFile();
+        }
+
+        // Drop a column and add a column; also delete 3 rows
+        BitSet rowsToDelete = new BitSet(5);
+        rowsToDelete.set(0);
+        rowsToDelete.set(1);
+        rowsToDelete.set(3);
+        info = createFileRewriter(useOptimizedOrcWriter).rewrite(
+                getColumnTypes(ImmutableList.of(7L, 13L, 18L), ImmutableList.of(createVarcharType(20), createVarcharType(5), INTEGER)),
+                newFile3,
+                newFile4,
+                rowsToDelete);
+        assertEquals(info.getRowCount(), 1);
```

Comment:
Any way to check if the remaining row is the one we want to keep? Or we can add this check in integration smoke test when testing `DELETE`.